infrastructure/zk/src/status.ts

```diff
@@ -0,0 +1,227 @@
+import { Command } from 'commander';
+
+import { Pool } from 'pg';
+import { ethers } from 'ethers';
+import { assert } from 'console';
+
+// Postgress connection pool - must be intialized later - as the ENV variables are set later.
+let pool: Pool | null = null;
+
+const GETTER_ABI = [
+    'function getTotalBlocksCommitted() view returns (uint256)',
+    'function getTotalBlocksVerified() view returns (uint256)',
+    'function getVerifierParams() view returns (bytes32, bytes32, bytes32)'
+];
+
+const VERIFIER_ABI = ['function verificationKeyHash() view returns (bytes32)'];
+
+export async function query(text: string, params?: any[]): Promise<any> {
+    const start = Date.now();
+
+    const res = await pool!.query(text, params);
+    const duration = Date.now() - start;
+    return res;
+}
+
+async function queryAndReturnRows(text: string, params?: any[]): Promise<any> {
+    const result = await query(text, params);
+    return result.rows;
+}
+
+async function getProofProgress(l1_batch_number: number) {
+    const result = await query('select * from prover_jobs_fri where l1_batch_number = $1 ', [l1_batch_number]);
+
+    let successful = 0;
+    let failed = 0;
+    let in_progress = 0;
+    let queued = 0;
+
+    result.rows.forEach((element: { status: string; error: string | undefined }) => {
+        if (element.status == 'successful') {
+            successful += 1;
+        } else {
+            if (element.error != null) {
+                failed += 1;
+            } else {
+                if (element.status == 'queued') {
+                    queued += 1;
+                }
+                if (element.status == 'in_progress') {
+                    in_progress += 1;
+                }
```

Comment:
double-checking: we have retries, so when job fails it gets re-queued, i.e. status gets changed to `queued` but the error is persisted in DB, so you'll count such jobs as `failed`. Is it what you expect to get?